packages/core/parcel-bundler/src/Asset.js

```diff
@@ -119,11 +119,12 @@ class Asset {
 
     this.addDependency(depName, Object.assign({dynamic: true, resolved}, opts));
 
-    const parsed = URL.parse(url);
-    parsed.pathname = this.options.parser
+    let parsed = URL.parse(url);
+    let name = this.options.parser
       .getAsset(resolved, this.options)
       .generateBundleName();
 
+    parsed.pathname = path.relative(path.dirname(this.relativeName), name);
```

Comment:
`name` is always just a hash+extension, so this `relative` call doesn't really do anything useful.

Here's a repo where it still isn't working (the error resurfaced I tried to fix the original relative PR): https://github.com/mischnic/2736-svg-import
In the dist root, a style file contains `url('../logo_color.78fe8740.svg');`.

(I think this should be `path.posix.relative` to not generate backslashes on Windows.)

Reply:
the test for CSS that I updated fails without this line...

Reply:
In the repo I linked above, this is the core problem:

Relative name of asset: `a/style.css`. A URL Dependency is in root, so `../logo_color.78fe8740.svg` seems correct.

Actually, `a/style.css` gets put into the root as well (does the Asset know this?) and now the `../` part is wrong.

Reply:
ahhh you're right. I guess we cannot correctly generate the relative path.